### PR TITLE
(feature, csharp): Generate gRPC core utilities

### PR DESCRIPTION
**Pull request by amckinney**
This adds support for gRPC core utility generation for the C# SDK generator, which includes the `RawGrpcClient` and `GrpcRequestOptions` types.

This also makes a few other improvements along the way, including:
1. Add a `Ternary` AST node.
2. Export the `ClassInstantiation`'s `Argument` type as an independent type (so it can be used elsewhere).
3. Update the `AsIs` file writer to recognize files that belong in the new `Core/Public` package name.
4. Add a `skipDefaultInitializer` option to prevent specific collection fields from having a default initializer.

### CODE REVIEW

File: generators/csharp/codegen/src/context/AbstractCsharpGeneratorContext.ts [hunk 1]
```diff
@@ -118,6 +124,16 @@ export abstract class AbstractCsharpGeneratorContext<
         return [this.getNamespace(), ...this.getChildNamespaceSegments(fernFilepath)];
     }
 
+    public getNamespaceFromProtobufFileOrThrow(protobufFile: ProtobufFile): string {
+        const namespace = protobufFile.options?.csharp?.namespace ?? undefined;
```
Comment (dcb6) [inline generators/csharp/codegen/src/context/AbstractCsharpGeneratorContext.ts:128]: nit (and am curious): would behavior be any differernt if `?? undefined` was omitted?

File: generators/csharp/codegen/src/project/CsharpProject.ts [hunk 2]
```diff
@@ -75,8 +80,22 @@ export class CsharpProject {
...
+        for (const filename of this.context.getCoreAsIsFiles()) {
+            this.coreFiles.push(
+                await this.createAsIsFile({
+                    filename,
+                    namespace: this.context.getCoreNamespace()
+                })
+            );
+        }
+
+        for (const filename of this.context.getPublicCoreAsIsFiles()) {
```
Comment (dcb6) [inline generators/csharp/codegen/src/project/CsharpProject.ts:92]: thanks for this!

File: generators/csharp/sdk/src/SdkGeneratorContext.ts [hunk 3]
```diff
@@ -87,6 +88,10 @@ export class SdkGeneratorContext extends AbstractCsharpGeneratorContext<SdkCusto
         ];
     }
 
+    public getPublicCoreAsIsFiles(): string[] {
+        return [AsIsFiles.GrpcRequestOptions];
```
Comment (dcb6) [inline generators/csharp/sdk/src/SdkGeneratorContext.ts:92]: should this also be conditional on `context.hasGrpcEndpoints()`?

**Review by dcb6 — APPROVED** [review]
beautiful 🤩

File: generators/csharp/codegen/src/context/AbstractCsharpGeneratorContext.ts (hunk 1 above)
Comment (amckinney) [reply generators/csharp/codegen/src/context/AbstractCsharpGeneratorContext.ts:128]: d'oh! Great catch - no difference in behavior. I'll update!

File: generators/csharp/sdk/src/SdkGeneratorContext.ts (hunk 3 above)
Comment (amckinney) [reply generators/csharp/sdk/src/SdkGeneratorContext.ts:92]: Yes definitely, great catch 👍

**Pull request merged**